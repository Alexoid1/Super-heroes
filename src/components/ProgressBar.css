.container {
    max-width: 400px;
    margin: 10px 0;
  }
  
.container .progressbar-container {
  position: relative;
  width: 90%;
  height: 20px;
  border: 1px solid #FFF;
  border-radius: 10px;
  display: flex;
  align-items: center;
  justify-content: center;
  overflow: hidden;
}

.progressbar-complete {
    position: absolute;
    left: 0;
    top: 0px;
    height: 100%;
    background-color: #dd5f25;
    border-radius: 10px;
    animation: g 2500ms infinite ease-in-out;
    z-index: 2;
}
  
      
.progress {
    z-index: 2;
}
  
  
@keyframes g {
  0% { background-color: #dd5f25; }
  50% { background-color: #607e9e; }
  100% { background-color: #dd5f25; }
}
  