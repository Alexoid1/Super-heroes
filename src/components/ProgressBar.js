import React from 'react';
import PropTypes from 'prop-types';
import './ProgressBar.css';

function ProgressBar({ pts }) {
  return (
    <div className="container">
      <div className="progressbar-container">
        <div className="progressbar-complete" style={{ width: `${pts}%` }}>
          <div className="progressbar-liquid" />
        </div>
        <span className="progress">
          {pts}
          pts
        </span>
      </div>
    </div>

  );
}

ProgressBar.propTypes = {
  pts: PropTypes.number.isRequired,
};

export default ProgressBar;
