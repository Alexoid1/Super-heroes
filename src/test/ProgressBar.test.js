import { Provider } from 'react-redux';
import store from '../reducers/index';
import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import ProgressBar from '../components/ProgressBar';

const renderer = new ShallowRenderer();

it('should render ProgressBar correctly', () => {
    renderer.render(<Provider store={store}>
                      <ProgressBar />
                    </Provider>); 
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});