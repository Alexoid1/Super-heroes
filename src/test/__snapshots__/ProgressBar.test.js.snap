// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ProgressBar correctly 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ProgressBar />
</Context.Provider>
`;
